src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStreamAsyncMemoryOverloads.Fixer.cs

```diff
@@ -0,0 +1,100 @@
+            ImmutableArray.Create(PreferStreamAsyncMemoryOverloads.RuleId);
+
+        public sealed override FixAllProvider GetFixAllProvider() =>
+            WellKnownFixAllProviders.BatchFixer;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            Document doc = context.Document;
+            CancellationToken ct = context.CancellationToken;
+            SyntaxNode root = await doc.GetSyntaxRootAsync(ct).ConfigureAwait(false);
+            if (root.FindNode(context.Span) is SyntaxNode node)
+            {
+                SemanticModel model = await doc.GetSemanticModelAsync(ct).ConfigureAwait(false);
+                if (model.GetOperation(node, ct) is IInvocationOperation invocation)
+                {
+                    string methodName = invocation.TargetMethod.Name;
+
+                    string title;
+                    if (methodName == "ReadAsync")
+                    {
+                        title = MicrosoftNetCoreAnalyzersResources.PreferStreamReadAsyncMemoryOverloadsTitle;
+                    }
+                    else if (methodName == "WriteAsync")
+                    {
+                        title = MicrosoftNetCoreAnalyzersResources.PreferStreamWriteAsyncMemoryOverloadsTitle;
+                    }
+                    else
+                    {
+                        return;
+                    }
+
+                    context.RegisterCodeFix(
+                        CodeAction.Create(
+                            title: title,
+                            createChangedDocument: c => FixInvocation(doc, root, invocation, methodName),
+                            equivalenceKey: MicrosoftNetCoreAnalyzersResources.PreferStreamAsyncMemoryOverloadsMessage),
+                        context.Diagnostics);
+                }
+            }
+        }
+
+        private static Task<Document> FixInvocation(Document doc, SyntaxNode root, IInvocationOperation invocation, string methodName)
+        {
+            SyntaxGenerator generator = SyntaxGenerator.GetGenerator(doc);
+
+            // The stream object
+            SyntaxNode instanceNode = invocation.Instance.Syntax;
+
+            // Need the byte array object so we can invoke its AsMemory() method
+            SyntaxNode bufferInstanceNode = invocation.Arguments[0].Value.Syntax; // byte[] buffer
+
+            // These arguments are not modified, just moved inside AsMemory
+            SyntaxNode offsetNode = invocation.Arguments[1].Syntax; // int offset
+            SyntaxNode countNode = invocation.Arguments[2].Syntax;  // int count
+
+            // Generate an invocation of the AsMemory() methdo from the byte array object
+            SyntaxNode asMemoryExtensionMethodNode = generator.IdentifierName("AsMemory");
+            SyntaxNode asMemoryExpressionNode = generator.MemberAccessExpression(bufferInstanceNode, asMemoryExtensionMethodNode);
```

Comment:
```suggestion
            SyntaxNode asMemoryExpressionNode = generator.MemberAccessExpression(bufferInstanceNode, "AsMemory");
```

Reply:
There should be a Roslyn analyzer that suggest this 😄